externals/kyuubi-spark-sql-engine/pom.xml

```diff
@@ -45,6 +45,13 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.kyuubi</groupId>
+            <artifactId>kyuubi-spark-lineage_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
In general, we do not make engine depend on any extension. If user want to enable one extension, just put the jar into engine classpath.

You can move the related test to lineage module.